packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -198,26 +195,28 @@ export class SingleThreadedWallet extends EventEmitter<EventEmitterType>
   }
 
   public async syncChannels(channelIds: Bytes32[]): Promise<MultipleChannelOutput> {
-    const results = await Promise.all(channelIds.map(channelId => this.syncChannel({channelId})));
-    return {
-      outbox: mergeOutgoing(results.flatMap(r => r.outbox)),
-      channelResults: mergeChannelResults(results.flatMap(r => r.channelResult)),
-    };
+    const response = WalletResponse.initialize();
+
+    await Promise.all(channelIds.map(channelId => this._syncChannel(channelId, response)));
+
+    return response.multipleChannelOutput();
   }
 
   public async syncChannel({channelId}: SyncChannelParams): Promise<SingleChannelOutput> {
+    const response = WalletResponse.initialize();
+    await this._syncChannel(channelId, response);
+    return response.singleChannelOutput();
+  }
+
+  private async _syncChannel(channelId: string, responseBuilder: ResponseBuilder): Promise<void> {
     const {states, channelState} = await this.store.getStates(channelId);
 
-    const {participants, myIndex} = channelState;
+    const {myIndex, participants} = channelState;
 
-    return {
-      outbox: createOutboxFor(channelId, myIndex, participants, {
-        walletVersion: WALLET_VERSION,
-        signedStates: states,
-        requests: [{type: 'GetChannel', channelId}],
-      }),
-      channelResult: ChannelState.toChannelResult(channelState),
-    };
+    states.forEach(s => responseBuilder.stateSigned(s, myIndex, channelId));
+
+    responseBuilder.requestGetChannel(channelId, myIndex, participants);
+    responseBuilder.channelUpdatedResult(ChannelState.toChannelResult(channelState));
```

Comment:
have you forgotten to `return responseBuilder.singleChannelOutput()` ?

Reply:
I think this is ok. `_syncChannel` modifies the `responseBuilder` that it was passed, and the caller will call either `response.singleChannelOutput()` or `response.multipleChannelOutput()` depending on whether it was `syncChannel` or `syncChannels`.

Reply:
nm, I see now that we are mutating the  `responseBuilder` that was passed in.